docs/proposals/admission_control_changes.md

```diff
@@ -0,0 +1,485 @@
+    1. api.GenerateName
+    1. strategy.Validate
+    1. validation.ValidateObjectMeta
+    1. strategy.Canonicalize
+  1. e.Storage.Create
+  1. e.AfterCreate
+  1. e.Decorator
+
+#### Proposed behavior
+
+This is the proposed new behavior:
+
+1. decode request
+1. admit.ApplyDefaults
+1. r.BeforeCreate
+  1. If strategy-based, call rest.BeforeCreate
+    1. strategy.PrepareForCreate
+    1. api.FillInObjectMetaSystemFields
+    1. api.GenerateName
+  1. Otherwise, just do whatever is in the rest.Creater
+1. r.Validate
+  1. If strategy-based, call rest.Validate
+    1. strategy.Validate
+    1. validation.ValidateObjectMeta
+    1. strategy.Canonicalize
+  1. Otherwise, just do whatever is in the rest.Creater
+1. admit.Validate
+1. admit.EnforceConstraints
+  1. e.g. for quota check
+1. r.Create
+  1. For strategy-based handlers:
+    1. e.Storage.Create
+    1. e.AfterCreate
+    1. e.Decorator
+  1. Otherwise, just do whatever is in the rest.Creater
+1. admit.Finish (name TBD)
+
+Notes:
+
+- admit.ApplyDefaults replaces admit.Admit and gives plugins a chance to apply
+  defaults to the decoded request object
+- r.BeforeCreate is the `rest.Creater` interface's new method for performing
+  logic after admission defaulting and before validation
+- r.Validate is the `rest.Creater` interface's new method for performing
+  validation
+- admit.Validate is a new admission phase that is executed after the
+  `rest.Creater`'s validation (plugins should not modify storage)
+- admit.EnforceConstraints is another new admission phase that is executed after
+  admit.Validate (plugins can modify storage)
+- admit.Finish (name TBD) is the final admission phase, running after the
+  `rest.Creater` has modified persistent storage
+
+
+### Update
+
+#### Current behavior
+
+1. decode request
+1. admit.Admit
+1. r.Update (the items below are part of the generic registry store / strategy-based storage, `pkg/registry/generic/registry/store.go`)
+  1. rest.BeforeUpdate
+    1. strategy.PrepareForUpdate
+    1. validation.ValidateObjectMetaUpdate
+    1. strategy.ValidateUpdate
+    1. strategy.Canonicalize
+  1. e.Storage.GuaranteedUpdate
+  1. e.AfterUpdate
+  1. e.Decorator
+
+#### Proposed behavior
+
+1. decode request
+1. ??? admit.ApplyDefaults ???
+1. r.BeforeUpdate
+  1. If strategy-based, call rest.BeforeUpdate
+    1. strategy.PrepareForUpdate
+  1. Otherwise, just do whatever is in the rest.Updater
+1. r.ValidateUpdate
+  1. If strategy-based, call rest.ValidateUpdate
+    1. validation.ValidateObjectMetaUpdate
+    1. strategy.ValidateUpdate
+    1. strategy.Canonicalize
+  1. Otherwise, just do whatever is in the rest.Updater
+1. admit.Validate
+1. admit.EnforceConstraints
+1. r.Update
+  1. For strategy-based handlers:
+    1. e.Storage.GuaranteedUpdate
+    1. e.AfterUpdate
+    1. e.Decorator
+  1. Otherwise, just do whatever is in the rest.Updater
+1. admit.Finish (name TBD)
+
+Notes:
+
+- Should we have admit.ApplyDefaults?
```

Comment:
i think so.  if you update an object and delete a value that was previously set, the defaulter needs the opportunity to assert (or re-assert) the default value.